core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -535,15 +535,14 @@ final class SmartConnectorColumnRDD(
       context.addTaskCompletionListener { _ =>
         logDebug(s"The txid going to be committed is $txId " + tableName)
 
-        if ((txId ne null) && !txId.equals("null")
-        /* && !(tx.asInstanceOf[TXStateProxy]).isClosed() */ ) {
+        //if ((txId ne null) && !txId.equals("null")) {
           val ps = conn.prepareStatement(s"call sys.COMMIT_SNAPSHOT_TXID(?)")
-          ps.setString(1, txId)
+          ps.setString(1, if (txId == null) "null" else txId)
```

Comment:
Perhaps its cleaner to set a proper null instead of token:
```scala
if (txId ne null) {
  ps.setString(1, txId)
} else {
  ps.setNull(1, Types.VARCHAR)
}
```